src/targets/gpu/fuse_mlir.cpp

```diff
@@ -191,39 +191,81 @@ struct find_mlir_op
         return {new_gemm_based_op, top_inputs};
     }
 
+    // Whitelist supported fusion options, including imposing type constraints
+    // for cases where MLIR only supports an operation (usually a pointwise function)
+    // on particular types.
+    bool is_pointwise_op_supported_by_mlir(const instruction& i) const
+    {
+        using type_t                                      = shape::type_t;
+        const auto& name                                  = i.name();
+        const auto result_type                            = i.get_shape().type();
+        const std::initializer_list<type_t> allowed_types = {type_t::float_type,
+                                                             type_t::half_type,
+                                                             type_t::int8_type,
+                                                             type_t::int32_type,
+                                                             type_t::bool_type};
+        // Preliminary type check.
+        if(not contains(allowed_types, result_type))
+        {
+            return false;
+        }
+        const std::initializer_list<std::string> any_type_ops = {"@literal", "@param", "@return"};
+        const std::initializer_list<std::string> no_bool_ops  = {"convolution",
+                                                                "quant_convolution",
+                                                                "dot",
+                                                                "quant_dot",
```

Comment:
This comment is non-blocking, but I believe the granularity of those can be improved. In a corner case, for example, we don't support int32 type operands for all those ops (yet it is valid migraphx IR).

BTW, should fp16 type conv computation go with the `convolution` op or `quant_convolution` op in migraphx and mlir respectively?

Reply:
I think fp16 is regular convolution, not the quantized one, unless we want to have fp16 returning fp32 values as an operation.

And true, it's probably not out of the question to be even more strict here and really whitelist the supported conv and dot types - should I do that in this PR?

Reply:
> I think fp16 is regular convolution, not the quantized one, unless we want to have fp16 returning fp32 values as an operation.

Agreed. Sounds like fp16 is a different model compared to fp32 entirely.

> to be even more strict here and really whitelist the supported conv and dot types - should I do that in this PR?

I think you can do that independently in another PR. I'm not concerned because I haven't really seen any model that would operate on those corner cases. As far as I understand, we and migraphx's support range overlap nicely. Probably worthwhile a ticket to be more defensive here as a reminder.